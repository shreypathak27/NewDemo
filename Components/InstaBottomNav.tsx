import { useNavigation } from '@react-navigation/native';
import React from 'react';
import { StyleSheet, TouchableOpacity, ImageBackground, View, Text, Button, Dimensions } from 'react-native';

function InstaBottomNav({ }) {
  const screenWidth = Dimensions.get('window').width;
  const navigation = useNavigation();

  return (
    <View style={{ flex: 1, justifyContent: 'flex-end' }}>
      <View style={{ width: screenWidth, flexDirection: 'row', alignItems: 'center', justifyContent: 'space-evenly', paddingBottom: 16, paddingTop: 10, backgroundColor: '#14171A', borderTopWidth: 1, borderTopColor: '#54525a' }}>
        {/* <TouchableOpacity onPress={() => navigation.navigate('Home1')}>
          <ImageBackground source={{ uri: 'https://cdn-icons-png.freepik.com/512/6821/6821151.png' }} style={styles.imageBackground}>
          </ImageBackground>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Home2')}>
          <ImageBackground source={{ uri: 'https://cdn-icons-png.freepik.com/512/6821/6821151.png' }} style={styles.imageBackground}>
          </ImageBackground>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Home3')}>
          <ImageBackground source={{ uri: 'https://cdn-icons-png.freepik.com/512/6821/6821151.png' }} style={styles.imageBackground}>
          </ImageBackground>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Home4')}>
          <ImageBackground source={{ uri: 'https://cdn-icons-png.freepik.com/512/6821/6821151.png' }} style={styles.imageBackground}>
          </ImageBackground>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Home5')}>
          <ImageBackground source={{ uri: 'https://cdn-icons-png.freepik.com/512/6821/6821151.png' }} style={styles.imageBackground}>
          </ImageBackground>
        </TouchableOpacity> */}
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  imageBackground: {
    width: 35,
    height: 35,
    alignItems: 'center',
    justifyContent: 'center',
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    color: "white",
    fontSize: 24,
    padding: 20,
  }
});

export default InstaBottomNav;
