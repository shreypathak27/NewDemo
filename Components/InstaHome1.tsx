import React, { createContext, useState, useContext } from 'react';
import { View, Text, Button, TextInput, StyleSheet, TouchableOpacity, Image, KeyboardAvoidingView, Platform } from 'react-native';
import { NavigationContainer, useNavigation } from '@react-navigation/native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Context1 } from '../App.tsx'


const HomeScreen1 = ({  }) => {
  const context = useContext(Context1)

  const [isPlay1, setIsPlay1] = useState(true);

  const handlePress1 = () => {
    setIsPlay1(!isPlay1);
  };

  const imageUri1 = isPlay1
    ? 'https://cdn.iconscout.com/icon/free/png-256/free-play-333-458838.png'
    : 'https://cdn-icons-png.flaticon.com/256/151/151859.png';

  const [isPlay2, setIsPlay2] = useState(true);

  const handlePress2 = () => {
    setIsPlay2(!isPlay2);
  };

  const imageUri2 = isPlay2
    ? 'https://cdn.iconscout.com/icon/free/png-256/free-play-333-458838.png'
    : 'https://cdn-icons-png.flaticon.com/256/151/151859.png';

  const [isPlay3, setIsPlay3] = useState(true);

  const handlePress3 = () => {
    setIsPlay3(!isPlay3);
  };

  const imageUri3 = isPlay3
    ? 'https://cdn.iconscout.com/icon/free/png-256/free-play-333-458838.png'
    : 'https://cdn-icons-png.flaticon.com/256/151/151859.png';

  const [isPlay4, setIsPlay4] = useState(true);

  const handlePress4 = () => {
    setIsPlay4(!isPlay4);
  };

  const imageUri4 = isPlay4
    ? 'https://cdn.iconscout.com/icon/free/png-256/free-play-333-458838.png'
    : 'https://cdn-icons-png.flaticon.com/256/151/151859.png';



  const handlePress5 = () => {
    context.setSongName("Stress Relieving Song")
  }
    
  const handlePress6 = () => {
    context.setSongName("Meditation Song")
  }


  return (
    <SafeAreaView style={styles.mainContainer}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={{ flex: 1 }}
      >
      <View style={styles.container}>
        <View>
          <Text style={styles.greetingText}>Hi Shrey!</Text>
        </View>
        <View style={styles.profileImage}></View>
      </View>
      <View style={styles.inputContainer}>
        <View style={styles.shadowContainer}>
          <TextInput style={styles.input} placeholder='Search...' placeholderTextColor='grey' />
        </View>
      </View>
      <View style={styles.cardContainer}>

          <TouchableOpacity onPress={handlePress5}>
        <View style={[styles.card, { backgroundColor: 'black' }]}>
          
          <Text style={styles.cardText}>Stress day Relaxation</Text>
          <View style={styles.rowContainer}>
            <TouchableOpacity onPress={handlePress1}>
              <View style={styles.buttonContainer}>
                <Image source={{ uri: imageUri1 }} style={{ width: 25, height: 25 }} />
              </View>
            </TouchableOpacity>
            <View style={styles.timeContainer}>
              <Text style={styles.buttonText}>15 mins</Text>
            </View>
          </View>
        </View>
        </TouchableOpacity>

          <TouchableOpacity onPress={handlePress6}>
        <View style={[styles.card, { backgroundColor: 'white' }]}>
          <Text style={[styles.cardText, { color: 'black' }]}>Evening Meditation to relax</Text>
          <View style={styles.rowContainer}>
            <TouchableOpacity onPress={handlePress2}>
              <View style={styles.buttonContainer}>
                <Image source={{ uri: imageUri2 }} style={{ width: 25, height: 25 }} />
              </View>
            </TouchableOpacity>
            <View style={[styles.timeContainer, { backgroundColor: '#ffedbd' }]}>
              <Text style={styles.buttonText}>12 mins</Text>
            </View>
          </View>
        </View>
        </TouchableOpacity>

      </View>
      <View style={styles.featuredContainer}>
        <Text style={styles.featuredText}>Featured for you</Text>
        <Text style={styles.seeAllText}>See all</Text>
      </View>
      <View style={styles.activityContainer}>
        <View style={styles.activityCard}>
          <View style={styles.activityContent}>
            <Text style={styles.activityTitle}>Explore new activities</Text>
            <View style={styles.activityRow}>
              <View style={styles.activityButton}>
                <Text>10 min</Text>
              </View>
              <View style={styles.activitySpacer} />
              <View style={[styles.activityButton, { borderColor: 'white', borderWidth: 1, backgroundColor: '#f7d474' }]}>
                <Text>Evening</Text>
              </View>
            </View>
          </View>
          <View style={styles.imageButtonContainer}>
            <TouchableOpacity onPress={handlePress3}>
              <View style={styles.imageButton}>
                <Image
                  source={{ uri: imageUri3 }}
                  style={{ width: 25, height: 25 }}
                />
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <View style={{ flex: 1 }} />
        <View style={[styles.activityCard, { backgroundColor: 'white' }]}>
          <View style={styles.activityContent}>
            <Text style={styles.activityTitle}>Meditation for sleep</Text>
            <View style={styles.activityRow}>
              <View style={[styles.activityButton, { backgroundColor: '#ffedbd' }]}>
                <Text>12 min</Text>
              </View>
              <View style={styles.activitySpacer} />
              <View style={[styles.activityButton, { borderColor: '#a8a8a8', borderWidth: 1, backgroundColor: 'white' }]}>
                <Text>Sleep</Text>
              </View>
              <View style={styles.activitySpacer} />
              <View style={[styles.activityButton, { borderColor: '#a8a8a8', borderWidth: 1, backgroundColor: 'white' }]}>
                <Text>Evening</Text>
              </View>
            </View>
          </View>
          <View style={styles.imageButtonContainer}>
            <TouchableOpacity onPress={handlePress4}>
              <View style={[styles.imageButton, { backgroundColor: '#f7d474' }]}>
                <Image
                  source={{ uri: imageUri4 }}
                  style={{ width: 25, height: 25 }}
                />
              </View>
            </TouchableOpacity>
          </View>
        </View>
        <View style={{ flex: 1 }} />
      </View>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1, 
    padding: 5, 
    paddingLeft: 25, 
    paddingRight: 25
  },

  input: {
    backgroundColor: "white",
    height: 60,
    borderRadius: 25,
    padding: 20,

  },
  container: {
    flex: 2,
    flexDirection: "row",
    justifyContent: 'space-around',
    alignItems: 'center'
  },
  greetingText: {
    fontSize: 30,
    paddingTop: 2,
    color: '#000'
  },
  profileImage: {
    height: 65,
    width: 65,
    borderRadius: 32.5,
    backgroundColor: 'grey'
  },
  inputContainer: {
    flex: 1,
    paddingBottom: 30,
  },
  shadowContainer: {
    shadowColor: 'grey',
    shadowOffset: { width: -2, height: 4 },
    shadowOpacity: 0.2,
    shadowRadius: 3,
  },
  cardContainer: {
    flex: 4,
    flexDirection: "row",
    justifyContent: 'space-between'
  },
  card: {
    width: 160,
    borderRadius: 25,
    padding: 15,
    justifyContent: 'space-between'
  },
  cardText: {
    color: "white",
    fontSize: 24,
  },
  rowContainer: {
    flexDirection: "row",
    justifyContent: 'space-evenly',
    alignItems: 'center'
  },
  buttonContainer: {
    height: 55,
    width: 55,
    borderRadius: 32.5,
    backgroundColor: '#f7d474',
    justifyContent: 'center',
    alignItems: 'center'
  },
  buttonText: {
    fontSize: 14,
    color: 'black'
  },
  timeContainer: {
    height: 35,
    width: 65,
    borderRadius: 32.5,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center'
  },
  featuredContainer: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: 10
  },
  featuredText: {
    color: '#474747',
    fontSize: 22
  },
  seeAllText: {
    color: '#a8a8a8',
    fontSize: 14,
    paddingTop: 3
  },
  activityContainer: {
    flex: 4,
  },
  activityCard: {
    flex: 12,
    backgroundColor: '#f7d474',
    borderRadius: 27,
    shadowColor: 'grey',
    shadowOffset: { width: -2, height: 4 },
    shadowOpacity: 0.2,
    shadowRadius: 3,
    flexDirection: 'row'
  },
  activityContent: {
    flex: 3,
    paddingTop: 5,
    paddingLeft: 20,
    justifyContent: 'space-around',
    padding: 5
  },
  activityTitle: {
    fontSize: 18
  },
  activityRow: {
    flexDirection: 'row'
  },
  activityButton: {
    height: 35,
    width: 65,
    borderRadius: 32.5,
    backgroundColor: 'white',
    alignItems: 'center',
    justifyContent: 'center'
  },
  activitySpacer: {
    width: 10
  },
  imageButtonContainer: {
    flex: 1,
    justifyContent: 'flex-end',
    paddingBottom: 10
  },
  imageButton: {
    height: 55,
    width: 55,
    borderRadius: 32.5,
    backgroundColor: 'white',
  
    justifyContent: 'center',
    alignItems: 'center'
  },
})

export default HomeScreen1;
