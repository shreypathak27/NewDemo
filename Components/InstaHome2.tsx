import React, { useContext, useEffect, useState } from 'react';
import { View, KeyboardAvoidingView, Platform, StyleSheet, Text, ActivityIndicator, FlatList } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Context1 } from '../App';
import { fetchTopTracks } from '../utility/LastFmApi';

interface Props {}

interface ContextType {
  songName: string; 
}

const HomeScreen2: React.FC<Props> = () => {
  const context = useContext(Context1);
  const songName  = context?.songName ?? 'No song';
  const [tracks, setTracks] = useState<any[]>([]);
    const [loading, setLoading] = useState<boolean>(true);

    useEffect(() => {
        const loadTopTracks = async () => {
            const topTracks = await fetchTopTracks('Cher');
            setTracks(topTracks);
            setLoading(false);
        };

        loadTopTracks();
    }, []);

  return (
    <SafeAreaView style={styles.mainContainer}>
      <KeyboardAvoidingView behavior={Platform.OS === 'ios' ? 'padding' : 'height'} style={styles.container}>
        <View style={styles.innerContainer}>
          <View style={styles.contentContainer}>
            <Text style={styles.title}>Currently playing: {songName}</Text>
          </View>
        </View>
        <View style={styles.container}>
            {loading ? (
                <ActivityIndicator size="large" color="#0000ff" />
            ) : (
                <FlatList
                    data={tracks}
                    keyExtractor={item => item.url}
                    renderItem={({ item }) => (
                        <Text>{item.name} by {item.artist.name}</Text>
                    )}
                />
            )}
        </View>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    padding: 5,
    paddingLeft: 25,
    paddingRight: 25,
    backgroundColor: "#eeeeee"
  },
  container: {
    flex: 1,
  },
  innerContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  contentContainer: {
    height: 300,
    width: 300,
    backgroundColor: 'black',
    borderRadius: 20,
    alignItems: 'center',
    justifyContent: 'center',
    padding: 20,
  },
  title: {
    color: 'white',
    fontSize: 20,
  },
});

export default HomeScreen2;
