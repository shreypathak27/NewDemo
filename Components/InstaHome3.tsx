import React, { useCallback, useContext, useMemo, useState } from 'react';
import { View, KeyboardAvoidingView, Platform, StyleSheet, Text } from 'react-native';
import { Button, TextInput } from 'react-native-paper';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useDispatch, useSelector } from 'react-redux';
import { deposit, withdraw } from '../balanceSlice';

function HomeScreen3() {
    
    const [number, setNumber] = useState(0)
    const doubleNumber = useMemo( () => {
        return doublethenumber(number)
    }, [number])

    function doublethenumber(num:number){
        console.log("Doubling up")
        return num*2
    }
  const [count, setCount] = useState(0);

  const increment = () => {
    console.log('Count going up!...');
    setCount((prevCount) => prevCount + 1);
  };

  const squaredCount = useMemo(() => {
    console.log('Calculating squared count...');
    return count * count;
  }, [count]);
  return (
    
    <SafeAreaView style={styles.mainContainer}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={{ flex: 1 }}>       
        <View style={{flex: 0.1}}></View>
      <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
      <TextInput
        keyboardType="numeric"
        onChangeText={(text) => {
          const parsedNumber = parseInt(text);
          setNumber(isNaN(parsedNumber) ? 0 : parsedNumber);
        }}
        placeholder="Enter a number"
        style={{ borderWidth: 1, padding: 10, margin: 10 }}
      />
      <Text>{doubleNumber}</Text>
      <Text>Count: {count}</Text>
      <Text>Squared Count: {squaredCount}</Text>
      <Button onPress={increment} > GO </Button>
    </View>
      </KeyboardAvoidingView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
    mainContainer: {
      flex: 1,
      padding: 5,
      paddingLeft: 25,
      paddingRight: 25,
      backgroundColor: "#eeeeee"
    }
    ,
    title: {
      fontSize: 20,
      fontWeight: 'bold',
    },
  });
  

export default HomeScreen3