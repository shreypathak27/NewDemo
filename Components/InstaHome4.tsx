import React, { useState } from 'react';
import { FlatList, ListRenderItem, View, KeyboardAvoidingView, Platform, StyleSheet, Text } from 'react-native';
import { Button, TextInput } from 'react-native-paper';
import { SafeAreaView } from 'react-native-safe-area-context';
import { useDispatch, useSelector } from 'react-redux';
import { deposit, withdraw } from '../balanceSlice';

interface Item {
  id: string;
  title: string;
}

const DATA: Item[] = [
  { id: '1', title: 'Item 1' },
  { id: '2', title: 'Item 2' },
  { id: '3', title: 'Item 3' },
  // Add more data items as needed
];

const HomeScreen4 = () => {
  const dispatch = useDispatch();
  const balance = useSelector((state) => {
    console.log('Bank processing...');
    return state.balance.value;
  });

  const [items, setItems] = useState<Item[]>(DATA);

  const renderItem: ListRenderItem<Item> = ({ item }) => (
    <View style={{ padding: 20 }}>
      <Text>{item.title}</Text>
    </View>
  );

  const keyExtractor = (item: Item) => item.id;

  const loadMoreData = () => {
    const newData = [
      ...items,
      { id: String(items.length + 1), title: `Item ${items.length + 1}` },
    ];
    setItems(newData);
  };


  return (
    <SafeAreaView style={styles.mainContainer}>
      <KeyboardAvoidingView
        behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
        style={{ flex: 1 }}>       
        <View style={{flex: 0.1}}></View>
    <Text style={styles.title}>Your Personal Bank</Text>
      <View style={{ marginVertical: 40 }}>
      <Button onPress={() => { dispatch(deposit(10));}} >Deposit 10$</Button>
        <Button onPress={() => { dispatch(withdraw(10));}} >Withdraw 10$</Button>
      </View>
      <View style={{ marginTop: 20 }}>
        <Text style={{ fontSize: 20 }}>Current Balance: {balance}$</Text>
      </View>
      <View style = {{ marginTop: 20}}/>
      <Text style ={styles.title}> List! </Text>
      <FlatList
      data={items}
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      onEndReached={loadMoreData}
      onEndReachedThreshold={0.5} // Load more data when 50% of the list is scrolled
      ListFooterComponent={() => (
        <View style={{ padding: 10 }}>
          <Button onPress={loadMoreData}>Load More</Button>
        </View>
      )}
    />
      </KeyboardAvoidingView>
      </SafeAreaView>

      
  );
}

export default HomeScreen4


const styles = StyleSheet.create({
  mainContainer: {
    flex: 1,
    padding: 5,
    paddingLeft: 25,
    paddingRight: 25,
    backgroundColor: "#eeeeee"
  }
  ,
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    
  },
});
