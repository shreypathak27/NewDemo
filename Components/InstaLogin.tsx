import { useNavigation } from '@react-navigation/native';
import React, { useState, useContext, useEffect } from 'react';
import { StyleSheet, View, TextInput, TouchableOpacity, Text, ActivityIndicator, Alert, Image } from 'react-native';

const InstaLogin = () => {
    const navigation = useNavigation();

  return (
    
    <View style={styles.container}>
      <View style={styles.logoContainer}>
        <Image source={{
              uri: 'https://icon-library.com/images/white-instagram-icon/white-instagram-icon-9.jpg',
            }} />
      </View>
      <TextInput
        autoCapitalize='none'
        placeholder="Email"
        placeholderTextColor="#ccc"
        style={styles.input}
      />
      <TextInput
        autoCapitalize='none'
        placeholder="Password"
        placeholderTextColor="#ccc"
        secureTextEntry
        style={styles.input}
      />
      {/* <TouchableOpacity style={styles.login} onPress={() => navigation.navigate('Home1')}>
        <Text style={styles.loginLabel}>Login</Text>
      </TouchableOpacity> */}
      <TouchableOpacity style={styles.register}>
        <Text style={styles.registerLabel}>Register</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    backgroundColor:'#14171A'
  },
  logoContainer: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    height:100,
    width:100
  },
  input: {
    borderColor: '#ccc',
    borderRadius: 8,
    borderWidth: 1,
    fontSize: 16,
    marginHorizontal: 24,
    marginVertical: 8,
    padding: 12,
  },
  login: {
    // backgroundColor: '#3B82F6',
    borderRadius: 8,
    fontSize: 16,
    marginHorizontal: 24,
    marginVertical: 8,
    padding: 16,
  },
  loginLabel: {
    color: '#fff',
    fontWeight: 'bold',
    textAlign: 'center',
    textTransform: 'uppercase',
  },
  register: {
    backgroundColor: '#fff',
    fontSize: 16,
    marginHorizontal: 24,
    marginVertical: 8,
    padding: 16,
  },
  registerLabel: {
    color: '#000',
    fontWeight: 'bold',
    textAlign: 'center',
    textTransform: 'uppercase'
  }
});

export default InstaLogin;