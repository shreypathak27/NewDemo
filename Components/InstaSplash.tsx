import React from 'react';
import { StyleSheet, Text, View, Image } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f2f2f2',
  },
  logoContainer: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  logo: {
    width: 75,
    height: 75,
  },
  titleContainer: {
    flex: 2,
    justifyContent: 'center',
    alignItems: 'center',
  },
  titleText: {
    fontSize: 24,
    fontWeight: 'bold',
  },
});

export const InstaSplash = () => {
  return (
    <View style={styles.container}>
      <View style={styles.logoContainer}>
        <Image
          style={styles.logo}
          source={{
            uri: 'https://cdn-icons-png.flaticon.com/512/2954/2954893.png',
          }}
        />
      </View>
      <View style={styles.titleContainer}>
        <Text style={styles.titleText}>Welcome!</Text>
      </View>
    </View>
  );
};
