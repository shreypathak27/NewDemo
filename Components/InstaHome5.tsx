import * as React from 'react';
import { View, Text, Button } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import InstaBottomNav from './InstaBottomNav';
import LinearGradient from 'react-native-linear-gradient';

function HomeScreen5() {
  return (
    <LinearGradient colors={['#14171A', '#000000']} style={{ flex: 1 }}>
      <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
        <Text style={{ color: "white" }}>Home Screen 5</Text>
        <InstaBottomNav />
      </View>
    </LinearGradient>
  );
}

export default HomeScreen5;