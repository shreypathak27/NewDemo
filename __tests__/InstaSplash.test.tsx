import React from 'react';
import renderer from 'react-test-renderer';
import { InstaSplash } from '../Components/InstaSplash';
import { FlatList, Text } from 'react-native';


test('renders correctly', () => {
  const tree = renderer.create(<InstaSplash />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders the Flatlist component', () => {
    const tree = renderer.create(
        <FlatList
            data={['Item1', 'Item2', 'Item3']}
            keyExtractor={item => item}
            renderItem={({ item }) => <Text>{item}</Text>}
        />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});