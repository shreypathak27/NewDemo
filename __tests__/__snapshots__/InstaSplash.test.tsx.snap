// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#f2f2f2",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://cdn-icons-png.flaticon.com/512/2954/2954893.png",
        }
      }
      style={
        {
          "height": 75,
          "width": 75,
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 2,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Welcome!
    </Text>
  </View>
</View>
`;

exports[`renders the Flatlist component 1`] = `
<RCTScrollView
  data={
    [
      "Item1",
      "Item2",
      "Item3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text>
        Item3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
